foster55f/Hotel-Overlook
# Testing

What's this PR do?
This PR update created the testing for Room.js and all of its properties.  This also created the testing for BookingCollection class and its 2 methods.  Also updated functionality for customer to match most of manager functionality in being able to book a room.

Where should the reviewer start?
Start at the Room.js to see all properties and move to Room-test.js after. Start at the BookingCollection.js to see all methods and move to BookingCollection-test.js after. User-deck.js has been updated to match most of functionality of manager.

How should this be manually tested?
I would pull the repo down and console.log to make sure data is being pulled correctly. All tests passed on my end.

Any background context you want to provide?
This update gets much of Room and BookingCollection class testing completed.

@brittanystoroz
@khalidwilliams
